base_address_parish/views/res_country_view.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+    <record id="view_res_country_parish_extended_form" model="ir.ui.view">
+        <field name="model">res.country</field>
+        <field name="inherit_id" ref="base.view_country_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//div[hasclass('oe_button_box')]" position="inside">
+                <button name="%(action_res_parish_tree)d"
```

Comment:
Parishes belongs to city, I think that you can add this action on res.city, and maybe add a specific menu to edit all parish records